lib/Loop.luau

```diff
@@ -382,13 +393,26 @@ function Loop:begin(events)
 						if profiling then
 							profiling[system] = nil
 						end
+
 						return
 					end
 
 					local fn = systemFn(system)
-					debug.profilebegin("system: " .. systemName(system))
-
-					local thread = coroutine.create(fn)
+					local name = systemName(system)
+
+					debug.profilebegin("system: " .. name)
+					local commitFailed = false
+					local thread = coroutine.create(function(...)
```

Comment:
This is unnecessary, instead of making commit a potentially fatal error, just make it return true/false on whether it could be merged without conflict.